tests/test_workers/test_tasks/test_gomod.py

```diff
@@ -29,13 +66,23 @@ def test_fetch_gomod_source(
     mock_bundle_dir,
     dep_replacements,
     expect_state_update,
+    pkg_config,
+    pkg_results,
     has_pkg_lvl_deps,
     sample_deps_replace,
     sample_package,
     sample_pkg_deps,
     sample_pkg_lvl_pkg,
     sample_env_vars,
 ):
+    def directory_present(*args, **kwargs):
+        mock_subpath = mock.Mock()
+        (subpath,) = args
+        mock_subpath.go_mod_file.exists.return_value = pkg_results["present"][subpath]
+        mock_subpath.relpath.return_value = pkg_results["relpath"][subpath]
+        mock_subpath.source_dir = pkg_results["sourcedir"][subpath]
```

Comment:
`pkg_results` feels too generic here, maybe `pkg_paths` or `pkg_files`?

Reply:
I could name it something like `request_results`. It is intended top be a dictionary of the expected output for test verification. I could not figure out how to explicitly have a static output without having it toed to the `pkg_config`.

Reply:
I see, it seemed like the values were used primarily for mocking setup, but they are used for checking the result as well. In that case I'm fine with the current name